### ISSUE TITLE
BUG: Browser is not restarting when using runner class with junit suite parameters

### ISSUE THREAD

**Issue by mrpet88** [serenity-bdd/serenity-core#3193]
Run sample project with `clean test -Dcucumber.filter.tags="@MyTest"`

When running with junit suite parameters:
```
@Suite
@IncludeEngines("cucumber")
@SelectClasspathResource("/features")
@ConfigurationParameter(key = PLUGIN_PROPERTY_NAME, value = "io.cucumber.core.plugin.SerenityReporterParallel,pretty,json:build/test-results/cucumber/frontend/global/results.json,timeline:build/test-results/timeline")
public class CucumberTestSuite {}
```
And with serenity.conf values:
```
  restart.browser.for.each = scenario
  use.unique.browser = false
```

**Then browser is not restarting.** 

**Observe**
that if tests run with the classic cucumber runner class (and with `useJUnitPlatform()` in gradle deactivated)
```
@RunWith(CucumberWithSerenity.class)
@CucumberOptions(
        plugin = {"pretty"},
        features = "src/test/resources/features"
)
```

**Then browser is restarting as expected.**

Repo to reproduce:
https://github.com/mrpet88/basicProjects/tree/main/src

Other info:
`serenityCoreVersion = '3.6.x'`

**Comment by wakaleo** [comment]
The sample project doesn't compile.

**Comment by wakaleo** [comment]
I'm not seeing this behaviour with 3.9.7 - could you check with that version?

**Comment by mrpet88** [comment]
Yes 3.9.7 works, but I also found out that it was also happening because the two scenarios I was trying to run had exactly the same name (`Searching for a term`)
After upgrading the serenity core and changed the names it worked. But is 3.9.7 version a stable version that I could use ?

**Comment by wakaleo** [comment]
Yes, 3.9.7 is stable.

**Comment by mrpet88** [comment]
@wakaleo since this is open. Is there a known issue with screenshots on failures and parallel execution? I have a reproducible example if you want I can give another example repo.

**Comment by wakaleo** [comment]
I am not aware of such an issue

**Comment by mrpet88** [comment]
On same repo -> https://github.com/mrpet88/serenity-cucumber-test-project

**Running one scenario:**
![image](https://github.com/serenity-bdd/serenity-core/assets/40420772/33da6248-c98a-41ca-9b6c-18ea74d4b4d1)

**Running on parallel:**
![image](https://github.com/serenity-bdd/serenity-core/assets/40420772/b8d6a6c2-23b8-4700-8658-f8f4b8654af1)

**Comment by wakaleo** [comment]
I won't have time to look at this soon: could you take a look and raise a PR?

**Comment by mrpet88** [comment]
I could try at least. Can you give me a small direction on where should I look for fixing?

**Comment by cliviu** [comment]
I can reproduce it, it looks like we have a bug, reported also here : https://github.com/serenity-bdd/serenity-core/issues/3176 . I will take a look

**Comment by cliviu** [comment]
@mrpet88 ,  I have pushed my changes here: https://github.com/cliviu/serenity-core/tree/screenshots_on_failures 
Let my know please if you can try it.

**Comment by mrpet88** [comment]
@cliviu is there a public snapshot version I could use? Or do I need to build the jar?

**Comment by cliviu** [comment]
Currently there is no publicversion

**Comment by mrpet88** [comment]
Are there somewhere instuctions on how to use the branch as dependency? Not sure if I am missing something here but cant seem to be able to build the project

**Comment by cliviu** [comment]
No problem, just wait for the next release

**Comment by mrpet88** [comment]
@cliviu should this work under new 4.0.1 version?
